src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,72 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
-    public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+    public String getSalaryInfo(String[] names, String[] data,
+                                String dateFrom, String dateTo) {
+        return getDrawResultSalaries(names, data, dateFrom, dateTo);
+    }
+
+    private String getDrawResultSalaries(String[] names, String[] data,
+                                         String dateFrom, String dateTo) {
+        StringBuilder result = new StringBuilder();
+
+        result.append("Report for period ").append(dateFrom).append(" - ")
+                .append(dateTo).append(System.lineSeparator());
+        for (int i = 0; i < names.length; i++) {
+            result.append(names[i]).append(" - ")
+                    .append(getSalarySumEmployee(names,data,dateFrom,dateTo)[i]);
```

Comment:
Please rework approach. Now for each name you call getSalarySumEmployee() method that runs logic that is taking all names and works with them. Why? Pass there just one name and returns result for this particluar name, not all of them

```suggestion
                    .append(getSalarySumEmployee(names[i], data, dateFrom, dateTo));
```